File: examples/rust/examples/sha2_256.rs
#![no_main]

use std::intrinsics::assigner_sha2_256;

use ark_pallas::Fq;

type BlockType = [Fq; 2];

#[circuit]
pub fn sha256_example(first_input_block: BlockType, second_input_block: BlockType) -> BlockType {
    // This will be arkworks function in the future.
    // All of the implementation details will be hidden from user.
    let hash_result = assigner_sha2_256(
        [first_input_block[0].0, first_input_block[1].0],
        [second_input_block[0].0, second_input_block[1].0],
    );

    [hash_result[0].into(), hash_result[1].into()]
}

File: examples/cpp/sha2_256.cpp
#include <nil/crypto3/hash/algorithm/hash.hpp>
#include <nil/crypto3/hash/sha2.hpp>

using namespace nil::crypto3;

[[circuit]] typename hashes::sha2<256>::block_type
    sha256_example(typename hashes::sha2<256>::block_type first_input_block,
                   typename hashes::sha2<256>::block_type second_input_block) {

    typename hashes::sha2<256>::block_type hash_result = hash<hashes::sha2<256>>(first_input_block, second_input_block);

    return hash_result;
}
